### DISCUSSION TITLE
Default layouts for different projects

### DISCUSSION THREAD

**Discussion by Rasmus-Bertell** [zellij-org/zellij#3203] [Q&A] [answered]
Hi,

I'm switching from tmux and I'm trying to figure out how to use layouts. Usually I have specific layouts I want for specific projects, is there a way to do it in Zellij with layouts? I read about session resurrection, but I want to start each day from a fresh workspace and there also seemed to be some problems resurrecting certain commands I need. I'm essentially looking for something akin to tmuxinator or tmuxp.

**Comment by giyany** [comment]
Hi there,

Did you check out the tutorials for layout yet? You can create different layouts and apply each to a session, for example with

``
$ zellij --layout layout_file.kdl
``

The tutorial is pretty good:
https://zellij.dev/tutorials/layouts/

and the docs have a lot of information too:
https://zellij.dev/documentation/layouts

If you want help with a specific command or setting, the Zellij Discord/Matrix are helpful for that.

**Reply by Rasmus-Bertell** [reply]
Yes, but it gets tedious fast to always write `zellij --layout project_1` and `zellij --layout project_n` and if I have the cwd defined in the layout file then I lose the ability to have language specific layouts, e.g. `cd projects/rust_1 && zellij --layout rust` or `cd projects/php_n && zellij --layout php`.

What I would ideally want is to have project selector which can be configured to use specific layouts for specific projects. I could then run it through fuzzy finder to easily start working.

**Reply by giyany** [reply]
Some of the functionalities that will be added in the next version will help you out:

https://github.com/zellij-org/zellij/pull/3200

**Reply by Rasmus-Bertell** [reply]
Nice, that will certainly help. I'll still have to write some sort of wrapper for zellij to achieve the desired outcome, but at least that should make it easier.

**Reply by Rasmus-Bertell** [reply]
After using Zellij for a while I can say that this is not quite what I was looking for. I have to open the welcome screen, write the session name, select the working directory (starting from root and not cwd for whatever reason), and select layout, before I can begin working on the project I want.

Ideally I would have a list of projects and they would have the layout and working directory defined already, like tmuxinator or tmuxp. I want frictionless switching between the projects I'm working on and at the time of writing this I think Zellij struggles with this.

I tried using resurrected sessions, but they don't play nicely with Laravel artisan commands so I gave up on that. Either way I like to start with fresh layouts when I turn my computer back on and not be reminded of yesterdays mess.

**Reply by giyany** [reply, marked as answer]
To have what you're looking for precisely you may want to hack it yourself.
There's a plugin that may help, however: https://github.com/vdbulcke/zellij-workspace

**Reply by bachirelkhoury** [reply]
@Rasmus-Bertell for that I use a [Justfile](https://github.com/casey/just) in each project, and have a set of recipes/commands the project needs. One of them is `just start` which calls zellij and launches that layout. Zellij layout and the justfile are unique per project and committed to get along with the project.

No need to remember anything.. just start

**Reply by Rasmus-Bertell** [reply]
The problem I have with that is that I would have to detach zellij, navigate to the new directory and then just start, when switching context.

**Comment by valyntyler** [comment]
for those using a custom layout is there any way to have per-project kdl files modify the default global layout? (e.g. spawn panes in it, etc.)
